### ISSUE TITLE
Elasticsearch sink not working

### ISSUE THREAD

**Issue by commixon** [kubernetes/heapster#1211]
I have setup an elk stack in its own namespace, aka `es`. And there is a service called: `elasticsearch.es.svc.cluster.local`.

For the sink option I simply use: `--sink=elasticsearch:http://elasticsearch.es.svc.cluster.local:9200". This does not work and i suppose it has to do with: 

```
    // Set the URL endpoints of the ES's nodes. Notice that when sniffing is
    // enabled, these URLs are used to initially sniff the cluster on startup.
    if len(opts["nodes"]) < 1 {
        return nil, fmt.Errorf("There is no node assigned for connecting ES cluster")
    }
```

In case I and the `?nodes...` param, then I get:

```
factory.go:74] Failed to create sink: failed to create ElasticSearch client: no Elasticsearch node available
```

I have also tried to play with the sniff, timeouts etc options, so I'm stuck for the time being and the elasticsearch sink does not work

(Question: the nodes paramater seems to be mandatory? According to the `if len(opts["nodes"]) < 1`? Why is that?)

**Comment by xyuanlu** [comment]
I think what you should do is --sink=elasticsearch:?nodes=http://elasticsearch.es.svc.cluster.local:9200 
You need to specify the "node" param.

**Comment by lacion** [comment]
i am having the same problem

E0726 18:20:17.526386       1 factory.go:74] Failed to create sink: failed to create ElasticSearch client: no Elasticsearch node available

--sink=elasticsearch:?nodes=http://foo.bar:9200&esUserName=foo&esUserSecret=bar

**Comment by rvrignaud** [comment]
I have same problem with or without specifying scheme.

**Comment by commixon** [comment]
In my case it worked by specifying the nodes parameter, but also using the latest version.

**Comment by rvrignaud** [comment]
I did succeed to send metric to elasticsearch using 1.2.beta0.
But it seems that events are not pushed and specifying index is not working.
Any clue on that ?

**Comment by rvrignaud** [comment]
Still same issue with 1.2.beta1.
Did anyone succeed in pushing events to ES and specifying index name ?

**Comment by douglasader** [comment]
Agreed with above comments, specifying index does not change it. Also metrics get pushed to index, but no events. Version 1.2.0, any progress on this issue. Also might be nice to be able to specify if you want metrics and events or only one of them.

**Comment by huangyuqi** [comment]
From v1.2.0-beta.0, the ES sink has been added to events. Any issues about this event ES sink? @douglasader

**Comment by douglasader** [comment]
Well I'm using 1.2.0, which is newer then 1.2.0-beta.0 AFAIK and events are not sent to ES, note however the index is setup on ES for events and metrics though, but only metrics gets populated with   actual entries.

**Comment by piosz** [comment]
cc @AlmogBaku

**Comment by AlmogBaku** [comment]
1. @commixon the original issue of requiring using the `?nodes=` notation was solved in #1260 which merged

2. @douglasader you can only run the right app if you want only metrics/events (`heapster` - the daemon that push metrics; `eventer` - the daemon that push events)

3. I tried to debug it and it seems like the index is changed at least with #1313 (which refactored big parts of the ES code)
